pkg/apiserver/resthandler.go

```diff
@@ -147,7 +147,20 @@ func getRequestOptions(req *restful.Request, scope RequestScope, kind string, su
 		newQuery[subpathKey] = []string{req.PathParameter("path")}
 		query = newQuery
 	}
-	return queryToObject(query, scope, kind)
+	versioned, err := scope.Creater.New(scope.ServerAPIVersion, kind)
```

Comment:
> Can we hold on using ServerAPIVersion in more places? I don't think it is correct to have multiple groups using "v1" types…

I'd actually like to have an explicit dependency expressed upon the internal types accepted by the API.  That means plumbing down Sets of types (or group,version,kind), then using the scheme to see whether the incoming object can be coerced to the expected type.

`ServerAPIVersion` isn't sufficient for the task and tends to operate a silent dependency on an external type.